packages/api/cms-api/src/page-tree/site-preview.resolver.ts

```diff
@@ -0,0 +1,36 @@
+import { Inject } from "@nestjs/common";
+import { Args, Query, Resolver } from "@nestjs/graphql";
+import { GraphQLJSONObject } from "graphql-scalars";
+import { SignJWT } from "jose";
+
+import { RequiredPermission } from "../user-permissions/decorators/required-permission.decorator";
+import { ContentScope } from "../user-permissions/interfaces/content-scope.interface";
+import { SITE_PREVIEW_CONFIG } from "./page-tree.constants";
+
+export type SitePreviewConfig = {
+    secret: string;
+};
+
+@Resolver()
+export class SitePreviewResolver {
+    constructor(@Inject(SITE_PREVIEW_CONFIG) private readonly config: SitePreviewConfig) {}
+
+    @Query(() => String)
+    @RequiredPermission("pageTree")
+    async sitePreviewJwt(
+        @Args("scope", { type: () => GraphQLJSONObject }) scope: ContentScope,
```

Comment:
Shouldn't this be PageTreeNodeScopeInput? Yes, this requires a factory, but the GraphQL schema would be more precise. What do you think?

Reply:
We're doing for all other page-tree-related queries however: https://github.com/vivid-planet/comet/blob/main/packages/admin/cms-admin/src/pages/pagesPage/createPagesQuery.ts#L16

Reply:
Currently the page tree doesn't support partial scopes, therefore the page tree scope is always the site scope. We could change this, but I'm not sure if this is ever needed 🤔

Reply:
I don't think we will ever need site-preview for users that don't have permission to the page-tree, re-using the pageTree permission is Ok imho.

And regarding the scope type: i think it is not worth the extra effort to have it typed correctly